credentials/src/main/kotlin/web5/sdk/credentials/StatusListCredential.kt

```diff
@@ -0,0 +1,317 @@
+package web5.sdk.credentials
+
+import com.danubetech.verifiablecredentials.CredentialSubject
+import io.ktor.client.HttpClient
+import io.ktor.client.engine.cio.CIO
+import io.ktor.client.plugins.ClientRequestException
+import io.ktor.client.plugins.ResponseException
+import io.ktor.client.request.get
+import io.ktor.client.statement.bodyAsText
+import io.ktor.http.isSuccess
+import java.io.ByteArrayInputStream
+import java.io.ByteArrayOutputStream
+import java.net.URI
+import java.util.Base64
+import java.util.BitSet
+import java.util.Date
+import java.util.zip.GZIPInputStream
+import java.util.zip.GZIPOutputStream
+
+/**
+ * Type alias representing the danubetech Status List 2021 Entry data model.
+ * This typealias simplifies the use of the [com.danubetech.verifiablecredentials.credentialstatus.StatusList2021Entry] class.
+ */
+public typealias StatusList2021Entry = com.danubetech.verifiablecredentials.credentialstatus.StatusList2021Entry
+
+/**
+ * Status purpose of a status list credential or a credential with a credential status.
+ */
+public enum class StatusPurpose {
+  REVOCATION,
+  SUSPENSION
+}
+
+/**
+ * The JSON property key for an encoded list.
+ */
+private const val ENCODED_LIST: String = "encodedList"
+
+/**
+ * The JSON property key for a status purpose.
+ */
+private const val STATUS_PURPOSE: String = "statusPurpose"
+
+/**
+ * `StatusListCredential` represents a digitally verifiable status list credential according to the
+ * [W3C Verifiable Credentials Status List v2021](https://www.w3.org/TR/vc-status-list/).
+ *
+ * When a status list is published, the result is a verifiable credential that encapsulates the status list.
+ *
+ */
+public object StatusListCredential {
+  /**
+   * Create a [StatusListCredential] with a specific purpose, e.g., for revocation.
+   *
+   * @param statusListCredentialId The id used for the resolvable path to the status list credential [String].
+   * @param issuer The issuer URI of the credential, as a [String].
+   * @param statusPurpose The status purpose of the status list cred, eg: revocation, as a [StatusPurpose].
+   * @param issuedCredentials The credentials to be included in the status list credential, eg: revoked credentials, list of type [VerifiableCredential].
```

Comment:
Seems like the passed in credentials MUST have the `credentialStatus` property present, is that right? If so, would it make sense to add validation?

Reply:
yup we have this check in 
prepareCredentialsForStatusList
      requireNotNull(vc.vcDataModel.credentialStatus) { "no credential status found in credential" }